### ISSUE TITLE
CSI-Snapshotter sidecar container not available as part of vSphere CSI Controller

### ISSUE THREAD

**Issue by srivallabandev** [rancher/rke2#5520]
Rancher version : 
2.6.x and 2.7.x

RKE2 Version: 
v1.27.10+rke2r1

Node(s) CPU architecture, OS, and Version: 
Linux xxxx 5.3.18-150300.59.144-default #1 SMP Tue Dec 5 15:20:50 UTC 2023 (5dc33fd) x86_64 x86_64 x86_64 GNU/Linux

Cluster Configuration:
1 master 2 workers

**Describe the bug:**
The vsphere-csi-controller pod running in the kube-system namespace doesn't have the csi-snapshotter side car container that orchestrates the creation of the volume snapshot. The VolumeSnapshot related CRDs are present. This side car container is one of the prerequsites mentioned in the Volume Snapshot official documentation.

**Steps To Reproduce:**
 Install RKE2 and check the vsphere-csi-controller pods, there wont be csi-snapshotter sidecar container

**Expected behavior:**
CSI snapshotter sidecar container to be part of vSphere CSI controller pods. since we are using the helm based deployment from Rancher , this should be part of the helm chart.

**Actual behavior:**
csi-snapshotter sidecar not part of the vSphere csi controller

**Additional context / logs:**
VolumeShapshot related CRDs are available from release 1.26.x but for it to work , CSI snapshotter side car container is necessary as per the CSI volume shapshot documentation. Rancher chart for the vSphere CSI doesn't have this sidecar container. we have added it manually to test the functionality

Screenshots
we added this sidecar manually to the vSphere CSI controller pod, image for reference
![image](https://github.com/rancher/rke2/assets/149671885/79b3cdbb-e9b3-45e2-b2e7-b7d6a6ceab51)

**Comment by brandond** [comment]
Moving this to the correct repo - I'm not sure if the sidecar comes from vsphere or what, but it will need to be added to the charts in vsphere-charts first before we can use it in rke2.
